alacritty/build.rs

```diff
@@ -13,6 +13,6 @@
 // limitations under the License.
 
 fn main() {
-    let hash = rustc_tools_util::get_commit_hash().expect("couldn't get commit hash");
+    let hash = rustc_tools_util::get_commit_hash().unwrap_or("".to_owned());
     println!("cargo:rustc-env=GIT_HASH={}", hash);
```

Comment:
I think I liked the idea of checking if the environment variable is set already before overwriting it.

Here's an idea for how this might look:

```rust
use std::env;

fn main() {
    match env::var("GIT_HASH") {
        Ok(_) => {},
        _ => {
            let hash = rustc_tools_util::get_commit_hash()
                                        .unwrap_or("".to_owned());
            println!("cargo:rustc-env=GIT_HASH={}", hash);
        },
    }
}
```

Reply:
> I think I liked the idea of checking if the environment variable is set already before overwriting it.

I don't think I do. What do we gain by that? Best case scenario we save ourselves some compile-time code, which is basically free relative to the complete compile time of Alacritty.

However, worst-case the hash might be outdated or contain completely garbage info (like the hash from a different project, set by the user manually). That could lead to some really annoying results when trying to troubleshoot bugs.

Since this is not a standardized environment variable set by git, I'd highly recommend against just assuming that when it's set, it's gonna be set properly.

Reply:
Fair enough, I was thinking from the last PR there was some desire to be able to set this manually when it's not inside a git repository. Though now I'm thinking I agree with you, if it's not in a git directory, the solution to getting it to build with git info would be to clone the repo.